docs/developing-contracts/examples/DAO.md

```diff
@@ -0,0 +1,196 @@
+    proposal_id: u128,
+    proposals: BTreeMap<u128, Proposal>,
+ 	approved_list: Vec<ActorId>,
+}
+```
+The params `admin`, `approved_token_program_id`, `period_duration`, `grace_period_length`, `dilution_bound`, `abort_window` are set when initializing a contract.
+
+`total_shares` - total shares across all members. Initially it is zero.
+
+`members` - members of the DAO.
+
+`member_by_delegate_key` - mapping the key responsible for submitting proposals and the member address.
+
+`proposal_id` - the index of the last proposal.
+
+`proposals` - all proposals (the proposal queue).
+
+`approved_list` - list of actors that are approved for joining the DAO.
+
+```rust
+ pub struct Proposal {
+    pub proposer: ActorId, /// - the member who submitted the proposal
+    pub applicant: ActorId, /// - the applicant who wishes to become a member 
+    pub shares_requested: u128, /// - the amount of shares the applicant is requesting
+    pub yes_votes: u128, /// - the total number of YES votes for that proposal
+    pub no_votes: u128, /// - the total number of NO votes for that proposal
+    pub quorum: u128, /// - a certain threshold of YES votes in order for the proposal to pass
+  pub is_membership_proposal: bool, /// - true if it is a membership proposal, false if it is a funding 	 proposal
+    pub processed: bool, /// - true if the proposal has already been processed
+    pub did_pass: bool, /// - true if the proposal has passed
+    pub cancelledcanceled: bool, /// - true if the proposal has been cancelledcanceled
+    pub aborted: bool, /// - true if the proposal has been aborted
+    pub token_tribute: u128, /// - the number of tokens offered for shares
+    pub details: String, /// - proposal details
+    pub starting_period: u64, /// - the start of the voting period
+    pub max_total_shares_at_yes_vote: u128, /// - the number of total shares that were detected at yes votes
+    pub votes_by_member: BTreeMap<H256, Vote>, /// - the votes on that proposal by each member
+}
+```
+
+```rust
+pub struct Member {
+    pub delegate_key: ActorId, /// - the key responsible for submitting proposals and voting ( by default it is equal to member address)
+    pub shares: u128, /// - the shares of that member
+    pub highest_index_yes_vote: u128, /// - the index of the highest proposal on which the members voted YES (that value is checked when user is going to leave the DAO)
+}
+``` 
+
+### Functions
+
+#### Initializes the DAO with initial parameters.
+
+```rust
+fn init_dao(
+        &mut self,
+        admin: &ActorId,
+        approved_token_program_id: &ActorId,
+        guild_bank_id: &ActorId,
+        voting_period_length: u64,
+        period_duration: u64,
+        grace_period_length: u64,
+        abort_window: u64,
+        dilution_bound: u128,
+    )
+```
+
+#### Adds actors to approved list. The added actors can deposit their tokens to DAO and a DAO member can submit a proposal of joining them DAO.
```

Comment:
Better to add descriptions in the body, not in the title?

Reply:
changed